DisguiseUnityRenderStream/Runtime/PipelineAbstraction/Presenter.cs

```diff
@@ -0,0 +1,263 @@
+        static Vector4 NoResize(Vector2 srcSize, Vector2 dstSize)
+        {
+            var scale = srcSize / dstSize;
+            var offset = CenterUVOffset(scale);
+            
+            return new Vector4(scale.x, scale.y, offset.x, offset.y);
+        }
+        
+        static Vector4 FitWidth(Vector2 srcSize, Vector2 dstSize)
+        {
+            var yScale = AspectRatio(srcSize) * AspectRatio(dstSize);
+            var yOffset = CenterUVOffset(yScale);
+            
+            return new Vector4(1f, yScale, 0f, yOffset);
+        }
+        
+        static Vector4 FitHeight(Vector2 srcSize, Vector2 dstSize)
+        {
+            var xScale = AspectRatio(srcSize) / AspectRatio(dstSize);
+            var xOffset = CenterUVOffset(xScale);
+            
+            return new Vector4(xScale, 1f, xOffset, 0f);
+        }
+        
+        static Vector4 Letterbox(Vector2 srcSize, Vector2 dstSize)
+        {
+            var scrAspect = AspectRatio(srcSize);
+            var dstAspect = AspectRatio(dstSize);
+
+            if (scrAspect > dstAspect)
+                return FitWidth(srcSize, dstSize);
+            else
+                return FitHeight(srcSize, dstSize);
+        }
+        
+        static Vector4 FillAspectRatio(Vector2 srcSize, Vector2 dstSize)
+        {
+            var scrAspect = AspectRatio(srcSize);
+            var dstAspect = AspectRatio(dstSize);
+
+            if (scrAspect < dstAspect)
+                return FitWidth(srcSize, dstSize);
+            else
+                return FitHeight(srcSize, dstSize);
+        }
+
+        static float AspectRatio(Vector2 size)
+        {
+            return size.x / size.y;
+        }
+
+        static float CenterUVOffset(float scale)
+        {
+            return (1f - scale) / 2f;
+        }
+        
+        static Vector2 CenterUVOffset(Vector2 scale)
+        {
+            return (Vector2.one - scale) / 2f;
+        }
+    }
+    
+    /// <summary>
+    /// <para>
+    /// Blits a texture to the local screen.
+    /// A number of strategies are available to handle the size and aspect ratio differences between the two surfaces.
+    /// </para>
+    /// 
+    /// <para>
+    /// <see cref="PresenterInput"/> is responsible for adjusting the <see cref="UnityEngine.EventSystems.EventSystem"/>
+    /// mouse coordinates to account for the blit.
+    /// </para>
+    ///
+    /// <remarks>Assumes that the local screen is the primary display. Modify this class for local multi-monitor specifics.</remarks>
+    /// </summary>
+    class Presenter : MonoBehaviour
+    {
+        const string k_profilerTag = "Disguise Presenter";
+        const string k_profilerClearTag = "Disguise Presenter Clear";
+
+        public PresenterStrategy.Strategy m_strategy = PresenterStrategy.Strategy.FillAspectRatio;
+        
+        /// <summary>
+        /// The texture to present. Can be any 2D texture.
+        /// </summary>
+        public RenderTexture m_source;
```

Comment:
Do you really want this member to be publicly set-able?

Reply:
Atm the Presenter public fields are independent of each other and can be modified every frame without creating problems. I don't have a specific design in mind to switch to, open to suggestions.

Reply:
As it's used right now, this can be
```
public RenderTexture Source { get; protected set; }
```
I don't see the point of exposing more public mutable fields than is necessary, because it will most likely bite you in the ass down the line.

Edit: My suggestion is still needlessly exposed. I don't even see the need for a public getter, for that matter.

Reply:
Why was this resolved? What changed?

Reply:
I converted to getter setters but forgot to change to protected. Fixed